activerecord/lib/active_record.rb

```diff
@@ -454,6 +454,15 @@ def self.eager_load!
     ActiveRecord::ConnectionAdapters.eager_load!
     ActiveRecord::Encryption.eager_load!
   end
+
+  # Explicitly closes all database connections.
+  # Most adapters send a packet to the server to signal the deconnection.
+  # Ideally applications should call this method prior to forking, if they don't
+  # connections will be discarded, meaning the server won't be notified and won't
+  # be able to reclaim resources immediately.
+  def self.disconnect!
+    ConnectionAdapters::PoolConfig.disconnect!
+  end
```

Comment:
My concern with this being `ActiveRecord.disconnect!` defined at the top level is the API is confusing. Disconnect what? Whereas `ActiveRecord.connection.disconnect!` is clearly disconnecting the connection. I think this should be in the connection handler like: `ActiveRecord::Base.connection_handler.disconnect!`. What do you think?

Reply:
> Whereas `ActiveRecord.connection.disconnect!` is clearly disconnecting the connection. 

Yeah, but it's also connecting first if you weren't already, which is annoyinh.

>  What do you think?

What about `disconnect_all!`? The usage is really to ask AR to close all its connections `ActiveRecord::Base.connection_handler.disconnect!` is much less clear whether it will close other pools etc.